assign0/README.md

```diff
@@ -0,0 +1,354 @@
+
+Your program should be able to emulate the following sample runs. *Make sure to activate your virtual environment before executing these lines of code!*
+
+```
+(cs41-env)$ python cheese.py
+Good morning. Welcome to the National Cheese Emporium!
+What would you like? Red Windsor
+I'm afraid we don't have any Red Windsor.
+What would you like? Lancashire
+I'm afraid we don't have any Lancashire.
+What would you like? cheddar
+I'm afraid we don't have any cheddar.
+What would you like? Cheddar
+We have Cheddar, yessir.
+
+(cs41-env)$ python cheese.py
+Good morning. Welcome to the National Cheese Emporium!
+What would you like? Red Windsor
+I'm afraid we don't have any Red Windsor.
+What would you like? cHeDdAr
+I'm afraid we don't have any cHeDdAr.
+What would you like? Have you in fact got any cheese here at all?
+We have 3 cheese(s)!
+Muenster
+Cheddar
+Red Leicester
+What would you like? exit
+I'm afraid we don't have any exit.
+What would you like? LET ME OUT
+I'm afraid we don't have any LET ME OUT.
+What would you like? Cheddar
+We have Cheddar, yessir.
+```
+
+For anything that is not detailed in the above specification, your program can behave in any way you'd like. For example, you can customize the prompts and messages.
+
+### Hints
+
+You can check that an element is contained in a collection by using the keyword `in`.
+
+### Submission
+
+Please submit a file called `cheese.py` which contains the code for this segment of the assignment.
+
+## Part 3: Chatbot
+
+In the third part of this assignment, you will implement a chatbot program which has a conversation with the user. There are just two requirements for this program:
+
+1. *Ask the user for input and print out responses*. The program should read input from the console and, based on the input, print out some response. You don't have to respond immediately after every input; the program could keep asking questions, just as long as it responds eventually.
+2. *Access stored data*. The program should access data stored in a text file over the course of its run, so that data from previous runs is accessible to users in future runs.
+
+To give you an idea of the scope and complexity of what we expect from this part of the assignment, below are two examples of types of submissions which would fulfill the requirements of this component.
+
+### Example 1: The Doors of Destiny
+
+A simple user authentication chatbot might take the role of a gatekeeper. The gatekeeper stops all travellers attempting to pass through the gate, and asks them for their name and passphrase. If the traveller provides a name and passphrase whch are contained in the gatekeeer's Book of Records, the traveller may proceed. If not, the traveller is denied entry.
+
+Upon being asked for a name and passphrase, each traveller has two options: they may (a) provide a valid name and passphrase, or (b) bribe the guard - which allows them to add a new name and passphrase to the Book of Records - for a small fee.
```

Comment:
@cooper-mj This is incredibly well-explained and I think a student reading this will clearly be able to tell that you're passionate and what this chatbot would look like.